src/tests/profiler/unittest/releaseondetach.cs

```diff
@@ -0,0 +1,94 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace Profiler.Tests
+{
+    public delegate void ProfilerCallback();
+
+    class ReleaseOnShutdown
+    {
+        static readonly Guid ReleaseOnShutdownGuid = new Guid("B8C47A29-9C1D-4EEA-ABA0-8E8B3E3B792E");
+
+        static volatile bool _profilerDone = false;
+
+        [DllImport("Profiler")]
+        private static extern void PassBoolToProfiler(IntPtr boolPtr);
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static void WasteTime()
+        {
+            // Give time for the profiler to detach
+            Console.WriteLine("Waiting for profiler to detach...");
+            bool profilerSetFlag = false;
+            for (int i = 0; i < 100; ++i)
+            {
+                Thread.Sleep(TimeSpan.FromSeconds(1));
```

Comment:
I'd recommend polling with far lower latency, say 1ms 100,000 times. No single sleep makes a huge difference on its own, but historically one of the biggest areas of unnecessary delay in running a test pass were tests that spent a lot of time sleeping. Usually that in turn was slow polling intervals, waiting N sec for events that usually happen within a few ms.